pages/profile/[username].tsx

```diff
@@ -0,0 +1,165 @@
+import { BaseContainer } from "@components/Page/BaseContainer";
+import { Helmet } from "react-helmet";
+import React, { useEffect, useRef, useState } from "react";
+import LayoutContainer from "@components/LayoutContainer";
+import Image from "next/image";
+import { Header } from "@components/Header";
+import Toggle from 'react-toggle'
+import { useRouter } from "next/router";
+import {  MessageEnum } from "@shared/enum";
+import { GetServerSidePropsContext } from "next";
+import { toast } from "react-toastify";
+import { useGetProfile } from "@services/profile/profileHooks";
+import { LoaderWrapper } from "@components/LoaderWrapper";
+import useToastHook from "@hooks/toast/useToastHook"
+import getConfig from "next/config";
+
+interface PageProps {
+    username?: string
+}
+
+const { publicRuntimeConfig } = getConfig()
+
+export default function Profile(props: PageProps) {
+    const { data, isLoading } = useGetProfile(props.username);
+    const [message, setMessage] = useState('')
+    const router = useRouter();
+    useEffect(() => {
+        if (data) {
+            localStorage.setItem(MessageEnum.targetUser, data.user_id);
+        }
+    }, [data])
+
+    const textAreaRef = useRef<HTMLTextAreaElement>(null)
+
+    useEffect(() => {
+        adjustTextareaHeight();
+    }, [message]);
+
+    const adjustTextareaHeight = () => {
+        if (textAreaRef.current) {
+            const lineHeight = parseFloat(getComputedStyle(textAreaRef.current).lineHeight);
+            textAreaRef.current.style.height = `${lineHeight}px`;
+            const lines = Math.floor(textAreaRef.current.scrollHeight / lineHeight);
+            const numLines = lines <= 3 ? lines : 3;
+            textAreaRef.current.style.height = `${numLines * lineHeight}px`;
+        }
+    };
+
+    const { copyToClipboardToast } = useToastHook()
+
+    const showDownloadToast = () => toast('Download the BetterSocial app now to create a profile and send non-anonymous messages', {
+        position: "bottom-center",
+        autoClose: 3000,
+        hideProgressBar: true,
+        closeOnClick: true,
+        pauseOnHover: true,
+        theme: "dark",
+        closeButton: false,
+        style: {
+            borderRadius: '8px',
+            width: '270px',
+            margin: 'auto',
+            marginBottom: '60px',
+            textAlign: 'center',
+            backgroundColor: 'rgba(0, 0, 0, 0.25)'
+        },
+    })
+
+    const onShareButtonClicked = () => {
+        navigator.clipboard.writeText(`${publicRuntimeConfig.DYNAMIC_LINK_DOMAIN}/u/${props.username}`)
+        copyToClipboardToast()
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The clipboard API can throw an error if the write operation fails for any reason (for example, due to insufficient permissions). It's good practice to handle these potential errors with a try-catch block.

```diff
- navigator.clipboard.writeText(`${publicRuntimeConfig.DYNAMIC_LINK_DOMAIN}/u/${props.username}`)
+ try {
+     navigator.clipboard.writeText(`${publicRuntimeConfig.DYNAMIC_LINK_DOMAIN}/u/${props.username}`)
+     copyToClipboardToast()
+ } catch (err) {
+     console.error('Failed to copy text: ', err)
+ }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->